modules/core/05-port/types/ibc_legacy_module.go

```diff
@@ -234,13 +234,32 @@ func (LegacyIBCModule) OnRecvPacket(
 }
 
 // OnAcknowledgementPacket implements the IBCModule interface
-func (LegacyIBCModule) OnAcknowledgementPacket(
+func (im *LegacyIBCModule) OnAcknowledgementPacket(
 	ctx sdk.Context,
 	channelVersion string,
 	packet channeltypes.Packet,
 	acknowledgement []byte,
 	relayer sdk.AccAddress,
 ) error {
+	for i := len(im.cbs) - 1; i >= 0; i-- {
```

Comment:
this ended up looking super clean with these 2 interfaces for wrapping, nice job!